kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/ControllerUtils.java

```diff
@@ -106,15 +104,19 @@ public static Response createCorrectVariant(String responseObj, HttpHeaders head
     public static Variant getVariant(HttpHeaders headers) {
         Variant v = RestEasy960Util.getVariant(headers);
         if( v == null ) {
-            v = defaultVariant;
+            v = Variant.mediaTypes(getMediaType(headers)).add().build().get(0);;
```

Comment:
This seems a convoluted way to do the same thing? But I'm probably missing the whole reason that you did this, which fixes the bug, I'm assuming. :)